src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1130,6 +1132,9 @@ private void retrieveChanges(Run build, GitClient git, TaskListener listener) th
         }
 
         for (RemoteConfig remoteRepository : repos) {
+            if (remoteRepository == repos.get(0) && redundantFetchCheck){
```

Comment:
Beware of using `==`. Is this intentional? Or do you actually means
```suggestion
            if (remoteRepository.equals(repos.get(0)) && redundantFetchCheck){
```
?

Reply:
My bad, I'm comparing two strings, I should use `equals`. Thanks.